test/tracer.spec.ts

```diff
@@ -489,6 +491,281 @@ describe('Autoblocks Tracer', () => {
+        evaluators: [new MyAsyncEvaluator()],
+      });
+
+      expect(traceId).toEqual('mock-trace-id');
+
+      expectPostRequest({
+        message: 'mock-message',
+        traceId: undefined,
+        timestamp,
+        properties: {
+          evaluations: [
+            {
+              id: mockUuid,
+              score: 0.9,
+              threshold: undefined,
+              metadata: undefined,
+              evaluatorExternalId: 'my-async-evaluator',
+            },
+          ],
+        },
+      });
+    });
+
+    it('handles multiple evaluators', async () => {
+      const tracer = new AutoblocksTracer('mock-ingestion-key');
+      const { traceId } = await tracer.sendEvent('mock-message', {
+        evaluators: [
+          new MyEvaluator(),
+          new MyAsyncEvaluator(),
+          new MyFullInfoEvaluator(),
+          new MyAsyncFullInfoEvaluator(),
+        ],
+      });
+
+      expect(traceId).toEqual('mock-trace-id');
+
+      expectPostRequest({
+        message: 'mock-message',
+        traceId: undefined,
+        timestamp,
+        properties: {
+          evaluations: [
+            {
+              id: mockUuid,
+              score: 0.9,
+              threshold: undefined,
+              metadata: undefined,
+              evaluatorExternalId: 'my-evaluator',
+            },
+            {
+              id: mockUuid,
+              score: 0.9,
+              threshold: undefined,
+              metadata: undefined,
+              evaluatorExternalId: 'my-async-evaluator',
+            },
+            {
+              id: mockUuid,
+              score: 0.9,
+              threshold: {
+                gt: 0,
+                lte: 1,
+              },
+              metadata: {
+                'some-metadata-key': 'some-metadata-value',
+              },
+              evaluatorExternalId: 'my-full-info-evaluator',
+            },
+            {
+              id: mockUuid,
+              score: 0.9,
+              threshold: {
+                gt: 0,
+                lte: 1,
+              },
+              metadata: {
+                'some-metadata-key': 'some-metadata-value',
+              },
+              evaluatorExternalId: 'my-async-full-info-evaluator',
+            },
+          ],
+        },
+      });
+    });
+
+    describe('errors', () => {
+      let runEvaluatorUnsafeSpy: jest.SpyInstance;
+
+      beforeEach(() => {
+        runEvaluatorUnsafeSpy = jest
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          .spyOn(AutoblocksTracer.prototype as any, 'runEvaluatorUnsafe');
+      });
+
+      afterEach(() => {
+        runEvaluatorUnsafeSpy.mockClear();
+      });
+
+      it('does not block if there is an evaluator error', async () => {
+        class ErrorEvaluator extends BaseEventEvaluator {
+          id = 'error-evaluator';
+
+          evaluateEvent(): Evaluation {
+            throw new Error('Something unexpected happened');
```

Comment:
this line is not being reached because you are making `runEvaluatorUnsafe` fail -- https://github.com/autoblocksai/javascript-sdk/pull/118/files#r1520071825

Reply:
This isn't true. 

In the beforeEach - I am just setting up the spy. That doen't mock it's implementation. I only set a mock in the test case I need to use the spy in - at which point the implementation is overridden. 

If you just run the two tests here the logs will spit out

```
npm run test --filter tracer

> @autoblocks/client@0.0.0-automated test
> TZ=UTC jest tracer

  console.error
    error-evaluator evaluator failed.  Error: Something unexpected happened
 ...ommit
 
  console.error
    Failed to execute evaluators.  Error: Brutal!
        ...omit...
```

I did it this way so I didn't have to add another describe just to isolate the spy mock - but I can change since this is confusing

Reply:
oh gotcha yeah you're right, thanks for the explanation. yeah it is a little subtle but feel free to leave as-is

Reply:
I updated it 👍